# psychencode


Gandal et al analyzed autism spectrum disorder, schizophrenia, and bipolar disorder across multiple levels of transcriptomic organizationâ€”gene expression, local splicing, transcript isoform expression, and coexpression networks for both protein-coding and noncoding genes to produce a quantitative, genome-wide resource. 
They performed TWAS based on 2,188 postmortem frontal and temporal cerebral cortex samples from 1,695 adults. RNA-sequencing reads were aligned to the GRCh37.p13 (hg19) reference genome. We generated a model using elastic-net weights released by Gandal et al.
More info on the study: https://science.sciencemag.org/content/362/6420/eaat8127.
The TWAS is available at http://resource.psychencode.org

Sabrina re-formatted the prediction models generated by Gandal et al into a sqlite database so that it's compatible with with PrediXcan software. 
The script used for that is here https://github.com/hakyimlab/psychencode/blob/master/analysis/generate_weights.Rmd. The model weights were stored in Predictdb format and can be accessed from [here](https://uchicago.app.box.com/s/du6f4z1zcgtn2v5gqms8kjajt1lsaprh).

Comparison of the different prediction models are compared [here](https://hakyimlab.github.io/psychencode/get_r2_LV.html). Short conclusion: elastic net performance is just fine. No need to use other models. We also have seen that BSLMM and BLUP can increase LD contamination so I wouldn't recommend those models even if prediction performance was better.






