<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sabrina-mi" />

<meta name="date" content="2020-06-22" />

<title>psychencode</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">psychencode</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">psychencode</h1>
<h4 class="author">sabrina-mi</h4>
<h4 class="date">2020-06-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-07-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>psychencode/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200622code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200622)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200622code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200622)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/sabrinami/Github/psychencode
</td>
<td style="text-align:left;">
.
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/sabrinami/Github/psychencode/data/PEC_TWAS_weights
</td>
<td style="text-align:left;">
data/PEC_TWAS_weights
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhakyimlabpsychencodetreea2d4866b002e971f015888c4f2f8e5370b5e2f4ctargetblanka2d4866a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/hakyimlab/psychencode/tree/a2d4866b002e971f015888c4f2f8e5370b5e2f4c" target="_blank">a2d4866</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomhakyimlabpsychencodetreea2d4866b002e971f015888c4f2f8e5370b5e2f4ctargetblanka2d4866a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/hakyimlab/psychencode/tree/a2d4866b002e971f015888c4f2f8e5370b5e2f4c" target="_blank">a2d4866</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  .DS_Store
    Untracked:  models/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/psychencode.Rmd</code>) and HTML (<code>docs/psychencode.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/psychencode/blob/a2d4866b002e971f015888c4f2f8e5370b5e2f4c/analysis/psychencode.Rmd" target="_blank">a2d4866</a>
</td>
<td>
sabrina-mi
</td>
<td>
2020-07-17
</td>
<td>
more detailed intros
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/psychencode/blob/643fdaa8abb8c0e30b422892cd03a2275462bee2/analysis/psychencode.Rmd" target="_blank">643fdaa</a>
</td>
<td>
sabrina-mi
</td>
<td>
2020-07-08
</td>
<td>
predixcan tests
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/psychencode/blob/463d1bdd95c0142011692ecbb8d71729ba5923e1/analysis/psychencode.Rmd" target="_blank">463d1bd</a>
</td>
<td>
sabrina-mi
</td>
<td>
2020-06-24
</td>
<td>
added definitions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/psychencode/463d1bdd95c0142011692ecbb8d71729ba5923e1/docs/psychencode.html" target="_blank">463d1bd</a>
</td>
<td>
sabrina-mi
</td>
<td>
2020-06-24
</td>
<td>
added definitions
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/hakyimlab/psychencode/blob/b51eff64fc3374c936218d4ebd3bebca2ab78300/analysis/psychencode.Rmd" target="_blank">b51eff6</a>
</td>
<td>
sabrina-mi
</td>
<td>
2020-06-24
</td>
<td>
new
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/hakyimlab/psychencode/b51eff64fc3374c936218d4ebd3bebca2ab78300/docs/psychencode.html" target="_blank">b51eff6</a>
</td>
<td>
sabrina-mi
</td>
<td>
2020-06-24
</td>
<td>
new
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Gandal et al analyzed autism spectrum disorder, schizophrenia, and bipolar disorder across multiple levels of transcriptomic organization—gene expression, local splicing, transcript isoform expression, and coexpression networks for both protein-coding and noncoding genes to produce a quantitative, genome-wide resource. They performed TWAS based on 2,188 postmortem frontal and temporal cerebral cortex samples from 1,695 adults. RNA-sequencing reads were aligned to the GRCh37.p13 (hg19) reference genome. We generated a model using elastic-net weights released by Gandal et al. More info on the study: <a href="https://science.sciencemag.org/content/362/6420/eaat8127" class="uri">https://science.sciencemag.org/content/362/6420/eaat8127</a>. The TWAS is available at <a href="http://resource.psychencode.org" class="uri">http://resource.psychencode.org</a></p>
</div>
<div id="definitions" class="section level1">
<h1>Definitions</h1>
<p>Define variables in the terminal:</p>
<pre class="bash"><code>export PRE=/Users/sabrinami/Github/psychencode
export CODE=$PRE/code
export DATA=$PRE/data
export OUTPUT=$PRE/output
export MODEL=$PRE/models</code></pre>
<p>Now in R:</p>
<pre class="r"><code>PRE=&quot;/Users/sabrinami/Github/psychencode&quot;
DATA=glue::glue(&quot;{PRE}/data&quot;)
OUTPUT=glue::glue(&quot;{PRE}/output&quot;)
MODEL=glue::glue(&quot;{PRE}/models&quot;)</code></pre>
</div>
<div id="download-data" class="section level1">
<h1>Download Data</h1>
<p>TWAS data can be downloaded at <a href="http://resource.psychencode.org" class="uri">http://resource.psychencode.org</a> Or download from terminal by running</p>
<pre class="bash"><code>cd $DATA
wget &quot;http://resource.psychencode.org/Datasets/Derived/PEC_TWAS_weights.tar.gz&quot;</code></pre>
<p>And unzip</p>
<pre class="bash"><code>tar -xvf PEC_TWAS_weights.tar.gz</code></pre>
</div>
<div id="load-file" class="section level1">
<h1>Load File</h1>
<p>When you open the PEC_TWAS_weights directory, there will be ~15,000 binary files. Each file contain information for a single gene. When loaded, an .RDat file contains snps (snp info), wgt.matrix (weights), and cv.performance (cross validation) tables. In the snps table, the first column is chromosome, the fourth is position, the fifth is effect allele, and the sixth is reference allele. In the wgt.matrix table, the rownames are the snp ids, and the columns are the weights derived from each method for each snp. In R, set the directory to PEC_TWAS_weights, then load the file:</p>
<pre class="r"><code>setwd(&quot;/Users/sabrinami/Github/psychencode/data/PEC_TWAS_weights&quot;)
load(&quot;ENSG00000000457.wgt.RDat&quot;)</code></pre>
<p>Now, the snps, wgt.matrix, and cv.performance are defined as variables.</p>
</div>
<div id="load-libraries" class="section level1">
<h1>Load Libraries</h1>
<p>Run in R:</p>
<pre class="r"><code>suppressPackageStartupMessages(library(RSQLite))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(data.table))</code></pre>
</div>
<div id="convert-file-to-dataframe" class="section level1">
<h1>Convert File to Dataframe</h1>
<p>make_df will load a file and store its data as a dataframe. This is only for a single gene, so later will be repeated for all genes. The input is the name of the .RDat file, and it returns returns dataframe with gene, position, chromosome, ref allele, eff allele, and non-zero enet weights.</p>
<pre class="r"><code>make_df &lt;- function(file) {
  load(file)  
  weights &lt;- data.frame(wgt.matrix) 
  snps &lt;- data.frame(snps) 
  rownames(weights) &lt;- c() 
  weights$gene &lt;- substr(file, 1, nchar(file) - 9)
  weights$chromosome &lt;- snps$V1 
  weights$position &lt;- snps$V4
  weights$ref_allele &lt;- snps$V5
  weights$eff_allele &lt;- snps$V6
  weights %&gt;% filter(enet != 0) %&gt;% select(gene, chromosome, position, ref_allele, eff_allele, enet)
}</code></pre>
</div>
<div id="make-weights-table" class="section level1">
<h1>Make Weights Table</h1>
<p>First, combine .RDat file names in a vector</p>
<pre class="r"><code>files &lt;- list.files(pattern = &quot;\\.RDat&quot;)</code></pre>
<p>The goal is to write tab delimited file with gene, chr, pos, ref, eff, and enet data for all genes in directory. Convert the first file in the vector to dataframe, then write it to a text file. And repeat for the remaining files, then append to the same text file.</p>
<pre class="r"><code>pre_weights = glue::glue(&quot;{OUTPUT}/pre_weights.txt&quot;)
write.table(make_df(files[1]), pre_weights, sep = &quot;\t&quot;, quote = FALSE, row.names = FALSE)
for(i in 2:length(files)) {
  write.table(make_df(files[i]), pre_weights, append = TRUE, sep = &quot;\t&quot;, quote = FALSE, col.names = FALSE, row.names = FALSE)
}</code></pre>
</div>
<div id="add-rsids" class="section level1">
<h1>Add rsIDs</h1>
<p>Following Yanyu’s recommendation, rsids were added to pre_weights.txt using her python script and a hg19 lookup table. Her script is here: <a href="https://github.com/liangyy/misc-tools/tree/master/annotate_snp_by_position" class="uri">https://github.com/liangyy/misc-tools/tree/master/annotate_snp_by_position</a> The lookup table, dbSNP150_list.txt, contains chromosome, position, ref, alt, rsid, and dbSNPBuildID. So the rsid for each snp is generated by matching the chromosome and position from psychencode models to lookup table. The output, weights_out.txt, will have gene, chr, pos, ref, eff, and rsid as new_id. In the terminal, run:</p>
<pre class="bash"><code>python3 $CODE/annotate_snp_by_position.py \
--input $OUTPUT/pre_weights.txt --chr_col 2 --pos_col 3 \
--lookup_table $DATA/dbSNP150_list.txt.gz --lookup_chr_col 1 --lookup_start_col 2 --lookup_end_col 2 --lookup_newid_col 5 --if_input_has_header 1 \
--out_txtgz $OUTPUT/weights_out.txt.gz</code></pre>
</div>
<div id="add-varids" class="section level1">
<h1>Add varIDs</h1>
<p>Read weights_out.txt in R, and define varID from chromosome, position, reference and effect alleles. (The RNA-seq was in hg19, so varID is defined in b37)</p>
<pre class="r"><code>weights &lt;- fread(&quot;weights_out.txt.gz&quot;)
weights$varID &lt;- paste(paste(&quot;chr&quot;, weights$chromosome, sep = &quot;&quot;), weights$position, weights$ref_allele, weights$eff_allele, &quot;b37&quot;, sep = &quot;_&quot;)
weights &lt;- weights %&gt;% select(gene, new_id, varID, ref_allele, eff_allele, enet) %&gt;% rename(weight = enet, rsid = new_id)</code></pre>
</div>
<div id="make-extra-table" class="section level1">
<h1>Make Extra Table</h1>
<p>Generate number of snps for each gene from the weights table. For now, include blank columns to match PrediXcan format (gene, genename, n.snps.in.model, pred.perf.R2, pred.perf.pval, pred.perf.qval)</p>
<pre class="r"><code>extra &lt;- weights %&gt;% group_by(gene) %&gt;% summarise(n.snps.in.model = n())
extra$genename &lt;- NA
extra$pred.perf.R2 &lt;- NA
extra$pred.perf.pval &lt;- NA
extra$pred.perf.qval &lt;- NA
extra &lt;- extra[c(1, 3, 2, 4, 5, 6)]</code></pre>
</div>
<div id="write-to-sqlite-database" class="section level1">
<h1>Write to SQLite Database</h1>
<p>Create database connection, and write the weights and extra tables to database.</p>
<pre class="r"><code>model_db = glue::glue(&quot;{MODEL}/psychencode_model/psychencode.db&quot;)
conn &lt;- dbConnect(RSQLite::SQLite(), model_db)
dbWriteTable(conn, &quot;weights&quot;, weights)
dbWriteTable(conn, &quot;extra&quot;, extra)</code></pre>
<p>To double check, confirm there is a weights and extra table, and show their contents.</p>
<pre class="r"><code>dbListTables(conn)
dbGetQuery(conn, &#39;SELECT * FROM weights&#39;) %&gt;% head
dbGetQuery(conn, &#39;SELECT * FROM extra&#39;) %&gt;% head</code></pre>
<p>Lastly, disconnect from database connection</p>
<pre class="r"><code>dbDisconnect(conn)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.6.2 Rcpp_1.0.3      rprojroot_1.3-2 digest_0.6.23  
 [5] later_1.0.0     R6_2.4.1        backports_1.1.5 git2r_0.27.1   
 [9] magrittr_1.5    evaluate_0.14   highr_0.8       stringi_1.4.5  
[13] rlang_0.4.2     fs_1.3.1        promises_1.1.0  whisker_0.4    
[17] rmarkdown_2.1   tools_3.6.2     stringr_1.4.0   glue_1.3.1     
[21] httpuv_1.5.3.1  xfun_0.12       yaml_2.2.0      compiler_3.6.2 
[25] htmltools_0.4.0 knitr_1.27     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
