---
title: "generate weights"
author: "sabrina-mi"
date: "2020-06-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Introduction
Gandal et al analyzed autism spectrum disorder, schizophrenia, and bipolar disorder across multiple levels of transcriptomic organizationâ€”gene expression, local splicing, transcript isoform expression, and coexpression networks for both protein-coding and noncoding genes to produce a quantitative, genome-wide resource. 
They performed TWAS based on 2,188 frontal and temporal cerebral cortex samples from 1,695 adults. RNA-sequencing reads were aligned to the GRCh37.p13 (hg19) reference genome. We generated a model using elastic-net weights released by Gandal et al.
More info on the study: https://science.sciencemag.org/content/362/6420/eaat8127.
The TWAS is available at http://resource.psychencode.org

# Definitions
Define variables in the terminal:
```{bash, eval=FALSE}
export PRE=/Users/sabrinami/Github/psychencode
export CODE=$PRE/code
export DATA=$PRE/data
export OUTPUT=$PRE/output
export MODEL=$PRE/models
```
Now in R:
```{r, eval=TRUE}
PRE="/Users/sabrinami/Github/psychencode"
DATA=glue::glue("{PRE}/data")
OUTPUT=glue::glue("{PRE}/output")
MODEL=glue::glue("{PRE}/models")
```

# Download Data
TWAS data can be downloaded at http://resource.psychencode.org
Or download from terminal by running

```{bash, eval=FALSE}
cd $DATA
wget "http://resource.psychencode.org/Datasets/Derived/PEC_TWAS_weights.tar.gz"
```
And unzip
```{bash, eval=FALSE}
tar -xvf PEC_TWAS_weights.tar.gz
```

# Load File
When you open the PEC_TWAS_weights directory, there will be ~15,000 binary files. Each file contain information for a single gene.
When loaded, an .RDat file contains snps (snp info), wgt.matrix (weights), and cv.performance (cross validation) tables.
In the snps table, the first column is chromosome, the fourth is position, the fifth is effect allele, and the sixth is reference allele.
In the wgt.matrix table, the rownames are the snp ids, and the columns are the weights derived from each method for each snp.

```{r, eval=TRUE}
setwd("/Users/sabrinami/Github/psychencode/data/")
load("ENSG00000000457.wgt.RDat")
```
Now, the snps, wgt.matrix, and cv.performance are defined as variables.

# Load Libraries
Run in R:
```{r, eval=FALSE}
suppressPackageStartupMessages(library(RSQLite))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(data.table))
```

# Convert File to Dataframe
make_df will load a file and store its data as a dataframe. This is only for a single gene, so later will be repeated for all genes.
The input is the name of the .RDat file, and it returns returns dataframe with gene, position, chromosome, ref allele, eff allele, and non-zero enet weights.

In most cases, FUSION data will have columns V5 and V6 as non-effect alleles and effect alleles, respectively. We recommend confirming the ref and alt alleles by checking with the UCSC genome browser. I looked at the first couple rows of the snps table:
```{r, eval= TRUE}
head(snps)
```

In the UCSC browser in hg19, the snp at position 169319286 has reference allele T and effect allele G, so we set accordingly in the following function:

```{r, eval=FALSE}
make_df <- function(file) {
  load(file)  
  weights <- data.frame(wgt.matrix) 
  snps <- data.frame(snps) 
  rownames(weights) <- c() 
  weights$gene <- substr(file, 1, nchar(file) - 9)
  weights$chromosome <- snps$V1 
  weights$position <- snps$V4
  weights$ref_allele <- snps$V6
  weights$eff_allele <- snps$V5
  weights %>% filter(enet != 0) %>% select(gene, chromosome, position, ref_allele, eff_allele, enet)
}
```

A thank you to Jake Gockley for pointing out our previous error in the make_df file regarding the ref and eff alleles.
# Make Weights Table
First, combine .RDat file names in a vector
```{r, eval=FALSE}
files <- list.files(pattern = "\\.RDat")
```
The goal is to write tab delimited file with gene, chr, pos, ref, eff, and enet data for all genes in directory. Convert the first file in the vector to dataframe, then write it to a text file. And repeat for the remaining files, then append to the same text file.
```{r, eval=FALSE}
pre_weights = glue::glue("{OUTPUT}/pre_weights.txt")
write.table(make_df(files[1]), pre_weights, sep = "\t", quote = FALSE, row.names = FALSE)
for(i in 2:length(files)) {
  write.table(make_df(files[i]), pre_weights, append = TRUE, sep = "\t", quote = FALSE, col.names = FALSE, row.names = FALSE)
}
```

# Add rsIDs
Following Yanyu's recommendation, rsids were added to `pre_weights.txt` using her python script and a hg19 lookup table. 
Her script is here: https://github.com/liangyy/misc-tools/tree/master/annotate_snp_by_position
The lookup table, [dbSNP150_list.txt](https://uchicago.box.com/s/twr1igkhpfbnz7n2mjqhpyaon47w1hzm), contains chromosome, position, ref, alt, rsid, and dbSNPBuildID. We also keep it in CRI, `/gpfs/data/im-lab/nas40t2/Data/References/dbSNP/dbSNP150_list.txt.gz`. So the rsid for each snp is generated by matching the chromosome and position from psychencode models to lookup table.
The output, `weights_out.txt`, will have gene, chr, pos, ref, eff, and rsid as new_id.
In the terminal, run:
```{bash, eval=FALSE}
python3 $CODE/annotate_snp_by_position.py \
--input $OUTPUT/pre_weights.txt --chr_col 2 --pos_col 3 \
--lookup_table $DATA/dbSNP150_list.txt.gz --lookup_chr_col 1 --lookup_start_col 2 --lookup_end_col 2 --lookup_newid_col 5 --if_input_has_header 1 \
--out_txtgz $OUTPUT/weights_out.txt.gz
```

# Add varIDs

Read `weights_out.txt` in R, and define varID from chromosome, position, reference and effect alleles. (The RNA-seq was in hg19, so varID is defined in b37)

```{r, eval=FALSE}
weights <- fread("weights_out.txt.gz")
weights$varID <- paste(paste("chr", weights$chromosome, sep = ""), weights$position, weights$ref_allele, weights$eff_allele, "b37", sep = "_")
weights <- weights %>% select(gene, new_id, varID, ref_allele, eff_allele, enet) %>% rename(weight = enet, rsid = new_id)
```

# Make Extra Table

Generate number of snps for each gene from the weights table. For now, include blank columns to match PrediXcan format (gene, genename, n.snps.in.model, pred.perf.R2, pred.perf.pval, pred.perf.qval)

```{r, eval=FALSE}
extra <- weights %>% group_by(gene) %>% summarise(n.snps.in.model = n())
extra$genename <- NA
extra$pred.perf.R2 <- NA
extra$pred.perf.pval <- NA
extra$pred.perf.qval <- NA
extra <- extra[c(1, 3, 2, 4, 5, 6)]
```

# Write to SQLite Database
Create database connection, and write the weights and extra tables to database.
```{r, eval=FALSE}
model_db = glue::glue("{MODEL}/psychencode_model/psychencode.db")
conn <- dbConnect(RSQLite::SQLite(), model_db)
dbWriteTable(conn, "weights", weights)
dbWriteTable(conn, "extra", extra)
```
To double check, confirm there is a weights and extra table, and show their contents.
```{r, eval=FALSE}
dbListTables(conn)
dbGetQuery(conn, 'SELECT * FROM weights') %>% head
dbGetQuery(conn, 'SELECT * FROM extra') %>% head
```
Lastly, disconnect from database connection
```{r, eval=FALSE}
dbDisconnect(conn)
```
