---
title: "PGC GWAS test"
output: html_document
editor_options:
  chunk_output_type: console
---
# Download Data
I used a GWAS of Schizophrenia to test S-PrediXcan on the model generated from Gandal et al.'s Cross-Disorder TWAS, on http://resource.psychencode.org, on against PredictDB models trained on GTEx v7 data, subsampled to use European samples. The psychencode model db was based on elastic-net weights calculated in FUSION. Here is how they were generated: https://hakyimlab.github.io/psychencode/psychencode.html. The covariance matrix was calulated on a 1000 Genomes reference panel here: file:///Users/sabrinami/Github/analysis-sabrina/covariance_1000G_ref/covariances.html

The GWAS was performed on unrelated genotyped individuals of European ancestry, with 33,426 cases and 54,065 controls. More info on the study: https://www.cell.com/cell/pdf/S0092-8674(18)30658-5.pdf?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867418306585%3Fshowall%3Dtrue. Download the GWAS: https://www.med.unc.edu/pgc/download-results/, filter by Schizophrenia, then Schizophrenia + Bipolar Disorder.
The GTEx models used elastic net, they can be downloaded by running `wget https://zenodo.org/record/3572799/files/GTEx-V7_HapMap-2017-11-29.tar.gz`. We compared the Whole Blood and Brain Cortex models to the psychencode model.


# Definitions
```{bash, eval=FALSE}
conda activate imlabtools
METAXCAN=/Users/sabrinami/Github/MetaXcan/software
MODEL=/Users/sabrinami/Github/psychencode/models
RESULTS=/Users/sabrinami/Github/psychencode/output/test_results
DATA=/Users/sabrinami/Desktop/psychencode_test_data
```

Similarly, in R, load the libraries, then set the same definitions.

```{r libraries}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(qqman))
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(RSQLite))
suppressPackageStartupMessages(library(UpSetR))
```


```{r definitions}
PRE="/Users/sabrinami/Github/psychencode"
DATA="/Users/sabrinami/Desktop/psychencode_test_data"
RESULTS=glue::glue("{PRE}/output/test_results")
CODE=glue::glue("{PRE}/code")
MODEL=glue::glue("{PRE}/models")
source(glue::glue("{CODE}/load_data_functions.R"))
source(glue::glue("{CODE}/plotting_utils_functions.R"))

gencode_df = load_gencode_df()
```

# Run S-PrediXcan 
Run S-PrediXcan on SCZ GWAS with psychencode model, and repeat for GTEx Brain Cortex and Whole Blood models.
```{bash, eval=FALSE}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/SCZvsCONT.sumstats \
--model_db_path $MODEL/psychencode_model/psychencode.db \
--covariance $MODEL/psychencode_model/psychencode.txt.gz \
--or_column OR \
--pvalue_column P \
--snp_column SNP \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/pgc_scz/SCZvsCONT_psychencode.csv
```

```{bash, eval=FALSE}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/SCZvsCONT.sumstats \
--model_db_path $MODEL/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_europeans_tw_0.5_signif.db \
--covariance $MODEL/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances.txt.gz \
--or_column OR \
--pvalue_column P \
--snp_column SNP \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/pgc_scz/SCZvsCONT_Brain_Cortex.csv
```

```{bash, eval=FALSE}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/SCZvsCONT.sumstats \
--model_db_path $MODEL/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_europeans_tw_0.5_signif.db \
--covariance $MODEL/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances.txt.gz \
--or_column OR \
--pvalue_column P \
--snp_column SNP \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/pgc_scz/SCZvsCONT_Whole_Blood.csv
```

# S-PrediXcan Results
Load the psychencode S-PrediXcan association results, and check for significant genes.
```{r}
spredixcan_association_psychencode = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/pgc_scz/SCZvsCONT_psychencode.csv"), gencode_df)
dim(spredixcan_association_psychencode)
significant_genes_psychencode <- spredixcan_association_psychencode %>% filter(pvalue < 0.05/nrow(spredixcan_association_psychencode)) %>% arrange(pvalue)
```

```{r}
spredixcan_association_Brain_Cortex = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/pgc_scz/SCZvsCONT_Brain_Cortex.csv"), gencode_df)
dim(spredixcan_association_Brain_Cortex)
significant_genes_Brain_Cortex <- spredixcan_association_Brain_Cortex %>% filter(pvalue < 0.05/nrow(spredixcan_association_Brain_Cortex)) %>% arrange(pvalue)
```

```{r}
spredixcan_association_Whole_Blood = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/pgc_scz/SCZvsCONT_Whole_Blood.csv"), gencode_df)
dim(spredixcan_association_Whole_Blood)
significant_genes_Whole_Blood <- spredixcan_association_Whole_Blood %>% filter(pvalue < 0.05/nrow(spredixcan_association_Whole_Blood)) %>% arrange(pvalue)
```

## Plot S-PrediXcan Association

We can plot the intersection of genes from the association output for each model. PsychENCODE has a much larger sample size, so reports many more genes compared to the GTEx models.
```{r}
genes <- list(Brain_Cortex = spredixcan_association_Brain_Cortex$gene,
              Whole_Blood = spredixcan_association_Whole_Blood$gene,
              Psychencode = spredixcan_association_psychencode$gene)
upset(fromList(genes), order.by = 'freq', empty.intersections = 'on')
```

When we plot the intersection of significant genes, there is nice overlap.
```{r}
significant_genes <- list(Brain_Cortex = significant_genes_Brain_Cortex$gene,
              Whole_Blood = significant_genes_Whole_Blood$gene,
              Psychencode = significant_genes_psychencode$gene)
upset(fromList(significant_genes), order.by = 'freq', empty.intersections = 'on')
```

For each of the models, we can make a histogram and Q-Q plot of the genes with their p-values, which confirm that all three find significant genes.
```{r}
spredixcan_association_psychencode %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20) + ggtitle("Distribution of P-value (PsychENCODE)")

gg_qqplot(spredixcan_association_psychencode$pvalue)
```

```{r}
spredixcan_association_Brain_Cortex %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20) + ggtitle("Distribution of P-value (GTEx Brain Cortex")

gg_qqplot(spredixcan_association_Brain_Cortex$pvalue)
```

```{r}
spredixcan_association_Whole_Blood %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20) + ggtitle("Distribution of P-value (GTEx Whole Blood")

gg_qqplot(spredixcan_association_Whole_Blood$pvalue)
```

We can plot the intersection of genes found significant in each model. There is overlap between all three, which is a good sign.
```{r}
significant_genes <- list(Brain_Cortex = significant_genes_Brain_Cortex$gene,
              Whole_Blood = significant_genes_Whole_Blood$gene,
              Psychencode = significant_genes_psychencode$gene)
upset(fromList(significant_genes), order.by = 'freq', empty.intersections = 'on')
```

Next, we can plot the distribution of z-scores for each model:
```{r}
zscore_psychencode <- data.frame("zscore" = spredixcan_association_psychencode$zscore, "model" = "psychENCODE")
zscore_Brain_Cortex <- data.frame("zscore" = spredixcan_association_Brain_Cortex$zscore, "model" = "Brain Cortex")
zscore_Whole_Blood <- data.frame("zscore" = spredixcan_association_Whole_Blood$zscore, "model" = "Whole Blood")
zscore <- rbind(zscore_Brain_Cortex, zscore_psychencode, zscore_Whole_Blood)
ggplot(zscore, aes(x=model, y= zscore)) + geom_violin() + geom_boxplot(width=.5) + ggtitle("Distribution of Association Z-score")
```

We also compare z-scores between each model. Ideally, the z-scores calculated from multiple model are similar for each gene, so they would follow the identity line. The z-scores between PsychENCODE and GTEx Brain Cortex appear more consistent than the other pairs, which could mean our model is reliable. First, plot the Brain Cortex and Psychencode z-scores:
```{r}
Brain_Cortex_psychencode_zscores = inner_join(spredixcan_association_Brain_Cortex, spredixcan_association_psychencode, by=c("gene"))
dim(Brain_Cortex_psychencode_zscores)
Brain_Cortex_psychencode_zscores %>% ggplot(aes(zscore.x, zscore.y)) + geom_point() + ggtitle("S-PrediXcan z-score") + xlab("GTex Brain Cortex") + ylab("PsychENCODE") + geom_abline(intercept = 0, slope = 1)
```

Whole Blood and Psychencode:
```{r}
Whole_Blood_psychencode_zscores = inner_join(spredixcan_association_Whole_Blood, spredixcan_association_psychencode, by=c("gene"))
dim(Whole_Blood_psychencode_zscores)
Whole_Blood_psychencode_zscores %>% ggplot(aes(zscore.x, zscore.y)) + geom_point() + ggtitle("S-PrediXcan z-score") + xlab("GTex Whole Blood") + ylab("PsychENCODE") + geom_abline(intercept = 0, slope = 1)
```
Whole Blood and Brain Cortex:

```{r}
Whole_Blood_Brain_Cortex_zscores = inner_join(spredixcan_association_Whole_Blood, spredixcan_association_Brain_Cortex, by=c("gene"))
dim(Whole_Blood_Brain_Cortex_zscores)
Whole_Blood_Brain_Cortex_zscores %>% ggplot(aes(zscore.x, zscore.y)) + geom_point() + ggtitle("S-PrediXcan z-score") + xlab("GTex Whole Blood") + ylab("GTEx Brain Cortex") + geom_abline(intercept = 0, slope = 1)
```

