---
title: "test_scz_clozuk_pgc"
author: "sabrina-mi"
date: "2020-07-21"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

We are testing our prediction model derived from a PsychENCODE TWAS, by comparing its S-PrediXcan association results with GTEx Brain Cortex and Whole Blood tissue models. We are using the Walters Group Schizophrenia GWAS.
# Definitions

```{bash, eval=FALSE}
conda activate imlabtools
METAXCAN=/Users/sabrinami/Github/MetaXcan/software
MODEL=/Users/sabrinami/Github/psychencode/models
RESULTS=/Users/sabrinami/Github/psychencode/output/test_results
DATA=/Users/sabrinami/Desktop/psychencode_test_data
```

Similarly, in R, load the libraries, then set the same definitions.
```{r libraries}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(qqman))
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(RSQLite))
suppressPackageStartupMessages(library(UpSetR))
```


```{r definitions}
PRE="/Users/sabrinami/Github/psychencode"
DATA="/Users/sabrinami/Desktop/psychencode_test_data"
RESULTS=glue::glue("{PRE}/output/test_results")
CODE=glue::glue("{PRE}/code")
MODEL=glue::glue("{PRE}/models")
source(glue::glue("{CODE}/load_data_functions.R"))
source(glue::glue("{CODE}/plotting_utils_functions.R"))

gencode_df = load_gencode_df()
```

# Download Data
The Walters Group in the Psychiatric Genomics Consortium released a Schizophrenia GWAS, from genome-wide genotype data from the UK (CLOZUK sample) and the PGC. There were 40,675 cases and 64,643 controls in the combined analysis.
More info on the study: https://www.nature.com/articles/s41588-018-0059-2#MOESM4
More info on the GWAS results: https://walters.psycm.cf.ac.uk 
To download the data, run.
```{bash, eval=FALSE}
cd $DATA
wget "https://walters.psycm.cf.ac.uk/clozuk_pgc2.meta.sumstats.txt.gz" --no-check-certificate
gunzip clozuk_pgc2.meta.sumstats.txt.gz
```

# Reformatting
The snps in the GWAS are in IMPUTE2 format, so it will need to be modified to match the prediction models' varID format.
## Load GWAS and Plot
First, we plot the GWAS.
```{r plot_gwas}
scz_GWAS = fread(glue::glue("{DATA}/clozuk_pgc2.meta.sumstats.txt.gz"), header=TRUE, sep="\t")
# manhattan(scz_GWAS, chr="CHR", bp="BP", snp="SNP", p="P" )
# gg_qqplot(scz_GWAS$P)
```

## Reformat GWAS

Modify the SNPS column to match model format, chr_pos_ref_alt_build. The varID column matches the psychencode model varID format, and varID_v7 will match GTEx v7.
```{r}
scz_GWAS_mod <- scz_GWAS %>% mutate(A1=toupper(A1), A2=toupper(A2), varID = paste(paste("chr", CHR, sep=""), BP, A1, A2, "b37", sep="_"), varID_v7 = paste(CHR, BP, A1, A2, "b37", sep="_"))
write.table(scz_GWAS_mod, glue::glue("{DATA}/clozuk_pgc2.meta.sumstats.out.txt"), quote=FALSE, row.names=FALSE, sep = "\t")
```

## Reformat Model Covariance
The GWAS is missing rsids, so the varIDs will be used to match snps in the GWAS to those in the models. The covariance files use rsids, which should be replaced with their varIDs. The weights table in the model has the rsid and varID for each snp, so this mapping can be used to swap in the covariance matrix.
First, open a connection to the model, then query the weights table. Load the covariance matrix.
```{r, eval=FALSE}
psychencode_model = glue::glue("{MODEL}/psychencode_model/psychencode.db")
conn <- dbConnect(RSQLite::SQLite(), psychencode_model)
snps <- dbGetQuery(conn, 'SELECT rsid, varID FROM weights')
snps_mapping <- distinct(snps)
dbDisconnect(conn)
```

Define the varId-rsid snp mapping using the unique snps in the weights table. Replace the RSID1 and RSID2 columns with left joins, then save the table.
```{r, eval=FALSE}
psychencode_covariance = fread(glue::glue("{MODEL}/psychencode_model/psychencode.txt.gz"), header=TRUE, sep=" ")
psychencode_covariance_mod <- psychencode_covariance %>% left_join(snps_mapping, by=c("RSID1"="rsid")) %>% select(GENE, varID, RSID2, VALUE) %>% rename(RSID1 = varID)
psychencode_covariance_mod <- psychencode_covariance_mod %>% left_join(snps_mapping, by=c("RSID2"="rsid")) %>% select(GENE, RSID1, varID, VALUE) %>% rename(RSID2 = varID)
write.table(psychencode_covariance_mod, glue::glue("{MODEL}/psychencode_model/psychencode_varID.txt"), quote=FALSE, row.names=FALSE)
```

Repeat for the GTEx models. 
```{r, eval=FALSE}
Brain_Cortex_model = glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_europeans_tw_0.5_signif.db")
conn <- dbConnect(RSQLite::SQLite(), Brain_Cortex_model)
snps <- dbGetQuery(conn, 'SELECT rsid, varID FROM weights')
snps_mapping <- distinct(snps)
dbDisconnect(conn)
```

```{r, eval=FALSE}
Brain_Cortex_covariance = fread(glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances.txt.gz"), header=TRUE, sep=" ")
Brain_Cortex_covariance_mod <- Brain_Cortex_covariance %>% left_join(snps_mapping, by=c("RSID1"="rsid")) %>% select(GENE, varID, RSID2, VALUE) %>% rename(RSID1 = varID)
Brain_Cortex_covariance_mod <- Brain_Cortex_covariance_mod %>% left_join(snps_mapping, by=c("RSID2"="rsid")) %>% select(GENE, RSID1, varID, VALUE) %>% rename(RSID2 = varID)
write.table(Brain_Cortex_covariance_mod, glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances_varID.txt"), quote=FALSE, row.names=FALSE)
```

```{r, eval=FALSE}
Whole_Blood_model = glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_europeans_tw_0.5_signif.db")
conn <- dbConnect(RSQLite::SQLite(), Whole_Blood_model)
snps <- dbGetQuery(conn, 'SELECT rsid, varID FROM weights')
snps_mapping <- distinct(snps)
dbDisconnect(conn)
```

```{r, eval=FALSE}
Whole_Blood_covariance = fread(glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances.txt.gz"), header=TRUE, sep=" ")
Whole_Blood_covariance_mod <- Whole_Blood_covariance %>% left_join(snps_mapping, by=c("RSID1"="rsid")) %>% select(GENE, varID, RSID2, VALUE) %>% rename(RSID1 = varID)
Whole_Blood_covariance_mod <- Whole_Blood_covariance_mod %>% left_join(snps_mapping, by=c("RSID2"="rsid")) %>% select(GENE, RSID1, varID, VALUE) %>% rename(RSID2 = varID)
write.table(Whole_Blood_covariance_mod, glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances_varID.txt"), quote=FALSE, row.names=FALSE)
```

Lastly, compress the new covariances files.
```{bash, eval=FALSE}
gzip $MODEL/psychencode_model/psychencode_varID.txt
gzip $MODEL/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances_varID.txt
gzip $MODEL/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances_varID.txt
```

# Run S-PrediXcan
Run S-PrediXcan on SCZ GWAS with psychencode model, and repeat for GTEx Brain Cortex and Whole Blood models. This used the varIDs instead of rsids to match snps in the GWAS and the models.

```{bash, eval=FALSE}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/clozuk_pgc2.meta.sumstats.out.txt \
--model_db_path $MODEL/psychencode_model/psychencode.db \
--covariance $MODEL/psychencode_model/psychencode_varID.txt.gz \
--keep_non_rsid --model_db_snp_key varID \
--or_column OR \
--pvalue_column P \
--snp_column varID \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_psychencode.csv
```

Repeat for GTEx models.
```{bash, eval=FALSE}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/clozuk_pgc2.meta.sumstats.out.txt \
--model_db_path $MODEL/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_europeans_tw_0.5_signif.db \
--covariance $MODEL/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances_varID.txt.gz \
--keep_non_rsid --remove_ens_version --model_db_snp_key varID \
--or_column OR \
--pvalue_column P \
--snp_column varID_v7 \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_Brain_Cortex.csv
```

```{bash, eval=FALSE}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/clozuk_pgc2.meta.sumstats.out.txt \
--model_db_path $MODEL/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_europeans_tw_0.5_signif.db \
--covariance $MODEL/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances_varID.txt.gz \
--keep_non_rsid --additional_output --model_db_snp_key varID \
--or_column OR \
--pvalue_column P \
--snp_column varID_v7 \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_Whole_Blood.csv
```

# S-PrediXcan Results
Load the psychencode S-PrediXcan association results, and check for significant genes.
```{r load psychencode association}
spredixcan_association_psychencode = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_psychencode.csv"), gencode_df)
dim(spredixcan_association_psychencode)
significant_genes_psychencode <- spredixcan_association_psychencode %>% filter(pvalue < 0.05/nrow(spredixcan_association_psychencode)) %>% arrange(pvalue)
```

Repeat for GTEx models.
```{r load brain cortex association}
spredixcan_association_Brain_Cortex = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_Brain_Cortex.csv"), gencode_df)
dim(spredixcan_association_Brain_Cortex)
significant_genes_Brain_Cortex <- spredixcan_association_Brain_Cortex %>% filter(pvalue < 0.05/nrow(spredixcan_association_Brain_Cortex)) %>% arrange(pvalue)
```

```{r load whole blood association}
spredixcan_association_Whole_Blood = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_Whole_Blood.csv"), gencode_df)
dim(spredixcan_association_Whole_Blood)
significant_genes_Whole_Blood <- spredixcan_association_Whole_Blood %>% filter(pvalue < 0.05/nrow(spredixcan_association_Whole_Blood)) %>% arrange(pvalue)
```

## Plot S-PrediXcan Association
For each of the models, we can make a histogram and Q-Q plot of the genes with their p-values, which confirm that all three find significant genes.
```{r compare significant genes}
significant_genes <- list(Brain_Cortex = significant_genes_Brain_Cortex$gene,
              Whole_Blood = significant_genes_Whole_Blood$gene,
              Psychencode = significant_genes_psychencode$gene)
upset(fromList(significant_genes), order.by = 'freq', empty.intersections = 'on')
```

However, Q-Q plots show that all three models have significant genes.
For each of the models, we can make a histogram and Q-Q plot of the genes with their p-values.
```{r plot psychencode}
spredixcan_association_psychencode %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association_psychencode$pvalue)
```

```{r plot brain cortex}
spredixcan_association_Brain_Cortex %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association_Brain_Cortex$pvalue)
```

```{r plot whole blood}
spredixcan_association_Whole_Blood %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association_Whole_Blood$pvalue)
```

Next, we can plot the distribution of z-scores for each model:
```{r zscore distribution}
zscore_psychencode <- data.frame("zscore" = spredixcan_association_psychencode$zscore, "model" = "psychENCODE")
zscore_Brain_Cortex <- data.frame("zscore" = spredixcan_association_Brain_Cortex$zscore, "model" = "Brain Cortex")
zscore_Whole_Blood <- data.frame("zscore" = spredixcan_association_Whole_Blood$zscore, "model" = "Whole Blood")
zscore <- rbind(zscore_Brain_Cortex, zscore_psychencode, zscore_Whole_Blood)
ggplot(zscore, aes(x=model, y= zscore)) + geom_violin() + geom_boxplot(width=.4) + ggtitle("Distribution of Association Z-score")
```

We also compare z-scores between each model. Ideally, the z-scores calculated from multiple model are similar for each gene, so they would follow the identity line. First, plot the Brain Cortex and Psychencode z-scores:
```{r compare brain cortex psychencode zscores}
Brain_Cortex_psychencode_zscores = inner_join(spredixcan_association_Brain_Cortex, spredixcan_association_psychencode, by=c("gene"))
dim(Brain_Cortex_psychencode_zscores)
Brain_Cortex_psychencode_zscores %>% ggplot(aes(zscore.x, zscore.y)) + geom_point() + ggtitle("S-PrediXcan z-score") + xlab("GTex Brain Cortex") + ylab("PsychENCODE") + geom_abline(intercept = 0, slope = 1)
```

Whole Blood and Psychencode:
```{r compare whole blood psychencode zscores}
Whole_Blood_psychencode_zscores = inner_join(spredixcan_association_Whole_Blood, spredixcan_association_psychencode, by=c("gene"))
dim(Whole_Blood_psychencode_zscores)
Whole_Blood_psychencode_zscores %>% ggplot(aes(zscore.x, zscore.y)) + geom_point() + ggtitle("S-PrediXcan z-score") + xlab("GTex Whole Blood") + ylab("PsychENCODE") + geom_abline(intercept = 0, slope = 1)
```
Whole Blood and Brain Cortex:

```{r compare whole blood brain cortex zscores}
Whole_Blood_Brain_Cortex_zscores = inner_join(spredixcan_association_Whole_Blood, spredixcan_association_Brain_Cortex, by=c("gene"))
dim(Whole_Blood_Brain_Cortex_zscores)
Whole_Blood_Brain_Cortex_zscores %>% ggplot(aes(zscore.x, zscore.y)) + geom_point() + ggtitle("S-PrediXcan z-score") + xlab("GTex Whole Blood") + ylab("GTEx Brain Cortex") + geom_abline(intercept = 0, slope = 1)
```

As a sanity check, we can also compare the association results from the PGC GWAS and CLOZUK+PGC GWAS.

```{r}
spredixcan_association_psychencode2 = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/pgc_scz/SCZvsCONT_psychencode.csv"), gencode_df)
dim(spredixcan_association_psychencode2)
significant_genes_psychencode2 <- spredixcan_association_psychencode2 %>% filter(pvalue < 0.05/nrow(spredixcan_association_psychencode2)) %>% arrange(pvalue)
```

```{r compare psychencode significant genes}
significant_genes_scz <- list(PsychENCODE_PGC =
                            significant_genes_psychencode2$gene,
                          PsychENCODE_CLOZUK_PGC=
                            significant_genes_psychencode$gene)
upset(fromList(significant_genes_scz), order.by = 'freq', empty.intersections = 'on')
```

```{r compare psychencode zscores}
psychencode_zscores = inner_join(spredixcan_association_psychencode2, spredixcan_association_psychencode, by=c("gene"))
dim(psychencode_zscores)
psychencode_zscores %>% ggplot(aes(zscore.x, zscore.y)) + geom_point() + ggtitle("S-PrediXcan z-score") + xlab("PsychENCODE PGC") + ylab("PsychENCODE PGC+CLOZUK") + geom_abline(intercept = 0, slope = 1)
```
