---
title: "Testing Psychencode Model on Schizophrenia GWAS"
author: "sabrina-mi"
date: "2020-07-10"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Definitions

```{bash definitions}
conda activate imlabtools
METAXCAN=/Users/sabrinami/Github/MetaXcan/software
MODEL=/Users/sabrinami/Github/psychencode/models
RESULTS=/Users/sabrinami/Github/psychencode/output/test_results
DATA=/Users/sabrinami/Desktop/psychencode_test_data
```

Similarly, in R, load the libraries, then set the same definitions
```{r libraries}
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(qqman))
suppressPackageStartupMessages(library(data.table))
suppressPackageStartupMessages(library(RSQLite))
```


```{r definitions}
PRE="/Users/sabrinami/Github/psychencode"
DATA="/Users/sabrinami/Desktop/psychencode_test_data"
RESULTS=glue::glue("{PRE}/output/test_results")
CODE=glue::glue("{PRE}/code")
MODEL=glue::glue("{PRE}/models")
source(glue::glue("{CODE}/load_data_functions.R"))
source(glue::glue("{CODE}/plotting_utils_functions.R"))

gencode_df = load_gencode_df()
```

# Download Data
More info on the study: https://www.nature.com/articles/s41588-018-0059-2#MOESM4
More info on the GWAS results: https://walters.psycm.cf.ac.uk

```{bash}
cd $DATA
wget "https://walters.psycm.cf.ac.uk/clozuk_pgc2.meta.sumstats.txt.gz" --no-check-certificate
gunzip clozuk_pgc2.meta.sumstats.txt.gz
```

# Reformat GWAS

## Load GWAS and Plot
First, we plot the GWAS.
```{r plot_gwas}
scz_GWAS = fread(glue::glue("{DATA}/clozuk_pgc2.meta.sumstats.txt"), header=TRUE, sep="\t")
manhattan(scz_GWAS, chr="CHR", bp="BP", snp="SNP", p="P" )
gg_qqplot(scz_GWAS$P)
```

## Reformat GWAS

Modify the SNPS column to match model format.
```{r}
scz_GWAS_mod <- scz_GWAS %>% mutate(A1=toupper(A1), A2=toupper(A2), varID = paste(paste("chr", CHR, sep=""), BP, A2, A1, "b37", sep="_"), varID_v7 = paste(CHR, BP, A2, A1, "b37", sep="_"))
write.table(scz_GWAS_mod, glue::glue("{DATA}/clozuk_pgc2.meta.sumstats.out.txt"), quote=FALSE, row.names=FALSE)
```

## Reformat Model Covariance
Load the model and covariance matrix
```{r}
psychencode_model = glue::glue("{MODEL}/psychencode_model/psychencode.db")
psychencode_covariance = fread(glue::glue("{MODEL}/psychencode_model/psychencode.txt.gz"), header=TRUE, sep=" ")
conn <- dbConnect(RSQLite::SQLite(), psychencode_model)
snps <- dbGetQuery(conn, 'SELECT rsid, varID FROM weights')
dbDisconnect(conn)
```

```{r}
snps_mapping <- distinct(snps)
psychencode_covariance_mod <- psychencode_covariance %>% left_join(snps_mapping, by=c("RSID1"="rsid")) %>% select(GENE, varID, RSID2, VALUE) %>% rename(RSID1 = varID)
psychencode_covariance_mod <- psychencode_covariance_mod %>% left_join(snps_mapping, by=c("RSID2"="rsid")) %>% select(GENE, RSID1, varID, VALUE) %>% rename(RSID2 = varID)
write.table(psychencode_covariance_mod, glue::glue("{MODEL}/psychencode_model/psychencode_varID.txt"), quote=FALSE, row.names=FALSE)
```

Repeat for the GTEx models. 
```{r}
Brain_Cortex_model = glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_europeans_tw_0.5_signif.db")
Brain_Cortex_covariance = fread(glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances.txt.gz"), header=TRUE, sep=" ")
conn <- dbConnect(RSQLite::SQLite(), Brain_Cortex_model)
snps <- dbGetQuery(conn, 'SELECT rsid, varID FROM weights')
dbDisconnect(conn)
```

```{r}
snps_mapping <- distinct(snps)
Brain_Cortex_covariance_mod <- Brain_Cortex_covariance %>% left_join(snps_mapping, by=c("RSID1"="rsid")) %>% select(GENE, varID, RSID2, VALUE) %>% rename(RSID1 = varID)
Brain_Cortex_covariance_mod <- Brain_Cortex_covariance_mod %>% left_join(snps_mapping, by=c("RSID2"="rsid")) %>% select(GENE, RSID1, varID, VALUE) %>% rename(RSID2 = varID)
write.table(Brain_Cortex_covariance_mod, glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances_varID.txt"), quote=FALSE, row.names=FALSE)
```

```{r}
Whole_Blood_model = glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_europeans_tw_0.5_signif.db")
Whole_Blood_covariance = fread(glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances.txt.gz"), header=TRUE, sep=" ")
conn <- dbConnect(RSQLite::SQLite(), Whole_Blood_model)
snps <- dbGetQuery(conn, 'SELECT rsid, varID FROM weights')
dbDisconnect(conn)
```

```{r}
snps_mapping <- distinct(snps)
Whole_Blood_covariance_mod <- Whole_Blood_covariance %>% left_join(snps_mapping, by=c("RSID1"="rsid")) %>% select(GENE, varID, RSID2, VALUE) %>% rename(RSID1 = varID)
Whole_Blood_covariance_mod <- Whole_Blood_covariance_mod %>% left_join(snps_mapping, by=c("RSID2"="rsid")) %>% select(GENE, RSID1, varID, VALUE) %>% rename(RSID2 = varID)
write.table(Whole_Blood_covariance_mod, glue::glue("{MODEL}/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances_varID.txt"), quote=FALSE, row.names=FALSE)
```

Then compress the new covariances matrices.
```{bash}
gzip $MODEL/psychencode_model/psychencode_varID.txt
gzip $MODEL/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances_varID.txt
gzip $MODEL/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances_varID.txt
```

# Run S-PrediXcan
Run S-PrediXcan on SCZ GWAS with psychencode model, and repeat for GTEx Brain Cortex and Whole Blood models.

```{bash}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/clozuk_pgc2.meta.sumstats.out.txt \
--model_db_path $MODEL/psychencode_model/psychencode.db \
--covariance $MODEL/psychencode_model/psychencode_varID.txt.gz \
--keep_non_rsid --additional_output --model_db_snp_key varID \
--or_column OR \
--pvalue_column P \
--snp_column varID \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_psychencode.csv
```

```{bash}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/clozuk_pgc2.meta.sumstats.out.txt \
--model_db_path $MODEL/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_europeans_tw_0.5_signif.db \
--covariance $MODEL/GTEx-V7-en/gtex_v7_Brain_Cortex_imputed_eur_covariances_varID.txt.gz \
--keep_non_rsid --additional_output --model_db_snp_key varID \
--or_column OR \
--pvalue_column P \
--snp_column varID_v7 \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_Brain_Cortex.csv
```

```{bash}
python3 $METAXCAN/SPrediXcan.py --gwas_file $DATA/clozuk_pgc2.meta.sumstats.out.txt \
--model_db_path $MODEL/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_europeans_tw_0.5_signif.db \
--covariance $MODEL/GTEx-V7-en/gtex_v7_Whole_Blood_imputed_eur_covariances_varID.txt.gz \
--keep_non_rsid --additional_output --model_db_snp_key varID \
--or_column OR \
--pvalue_column P \
--snp_column varID_v7 \
--non_effect_allele_column A2 \
--effect_allele_column A1 \
--throw \
--output_file $RESULTS/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_Whole_Blood.csv
```

# S-PrediXcan Results
```{r}
spredixcan_association_psychencode = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_psychencode.csv"), gencode_df)
dim(spredixcan_association_psychencode)
significant_genes_psychencode <- spredixcan_association_psychencode %>% filter(pvalue < 0.05/nrow(spredixcan_association_psychencode)) %>% arrange(pvalue)
```

```{r}
spredixcan_association_Brain_Cortex = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_Brain_Cortex.csv"), gencode_df)
dim(spredixcan_association_Brain_Cortex)
significant_genes_Brain_Cortex <- spredixcan_association_Brain_Cortex %>% filter(pvalue < 0.05/nrow(spredixcan_association_Brain_Cortex)) %>% arrange(pvalue)
```

```{r}
spredixcan_association_Whole_Blood = load_spredixcan_association(glue::glue("{RESULTS}/spredixcan/eqtl/clozuk_pgc2/clozuk_pgc2_Whole_Blood.csv"), gencode_df)
dim(spredixcan_association_Whole_Blood)
significant_genes_Whole_Blood <- spredixcan_association_Whole_Blood %>% filter(pvalue < 0.05/nrow(spredixcan_association_Whole_Blood)) %>% arrange(pvalue)
```

## Plot S-PrediXcan Association
```{r}
spredixcan_association_psychencode %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association_psychencode$pvalue)
```

```{r}
spredixcan_association_Brain_Cortex %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association_Brain_Cortex$pvalue)
```

```{r}
Brain_Cortex_psychencode_association = inner_join(spredixcan_association_Brain_Cortex, spredixcan_association_psychencode, by=c("gene"))
dim(Brain_Cortex_psychencode_association)
Brain_Cortex_psychencode_association %>% ggplot(aes(zscore.x, zscore.y)) + geom_point()
```

```{r}
spredixcan_association_Whole_Blood %>% arrange(pvalue) %>% ggplot(aes(pvalue)) + geom_histogram(bins=20)

gg_qqplot(spredixcan_association_Whole_Blood$pvalue)
```
