---
title: "weights_from_best_performance_analysis"
author: "sabrina-mi"
date: "2020-08-04"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
# Definitions
```{r}
PRE="/Users/sabrinami/Github/psychencode"
DATA=glue::glue("{PRE}/data/PEC_TWAS_weights")
OUTPUT=glue::glue("{PRE}/output")
MODEL=glue::glue("{PRE}/models")
```

```{r}
setwd(DATA)
load("ENSG00000000457.wgt.RDat")
```

# Load Libraries
Run in R:
```{r}
suppressPackageStartupMessages(library(RSQLite))
suppressPackageStartupMessages(library(tidyverse))
suppressPackageStartupMessages(library(data.table))
```

# Convert File to Dataframe
make_df will load a file and store its data as a dataframe. This is only for a single gene, so later will be repeated for all genes.
The input is the name of the .RDat file, and it returns returns dataframe with gene, position, chromosome, ref allele, eff allele, and non-zero enet weights.

```{r}
make_df <- function(file) {
  load(file)
  best_performing_model <- colnames(cv.performance)[which.min(cv.performance["pval", ])]
  weights <- snps %>% rename(chromosome = V1, position = V4, ref_allele = V5, eff_allele = V6) %>% select(-c(V2, V3))
  weights$gene <- substr(file, 1, nchar(file) - 9)
  weights$weight <- wgt.matrix[ , best_performing_model]
  weights %>% filter(weight != 0)
}
```

# Make Weights Table
First, combine .RDat file names in a vector
```{r}
files <- list.files(pattern = "\\.RDat")
models <- data.frame(best_model = c(colnames(cv.performance)[which.min(cv.performance["pval", ])]))
for(i in 2:length(files)) {
  load(files[i])
  best_performing_model <- 
    data.frame(best_model =
                 c(colnames(cv.performance)[which.min(cv.performance["pval", ])]))
  models <- rbind(models, best_performing_model)
}
```
The goal is to write tab delimited file with gene, chr, pos, ref, eff, and enet data for all genes in directory. Convert the first file in the vector to dataframe, then write it to a text file. And repeat for the remaining files, then append to the same text file.

```{r}
pre_weights = glue::glue("{OUTPUT}/pre_weights.txt")
write.table(make_df(files[1]), pre_weights, sep = "\t", quote = FALSE, row.names = FALSE)
for(i in 2:length(files)) {
  write.table(make_df(files[i]), pre_weights, append = TRUE, sep = "\t", quote = FALSE, col.names = FALSE, row.names = FALSE)
}
```

# Add rsIDs
Following Yanyu's recommendation, rsids were added to pre_weights.txt using her python script and a hg19 lookup table. 
Her script is here: https://github.com/liangyy/misc-tools/tree/master/annotate_snp_by_position
The lookup table, dbSNP150_list.txt, contains chromosome, position, ref, alt, rsid, and dbSNPBuildID. So the rsid for each snp is generated by matching the chromosome and position from psychencode models to lookup table.
The output, weights_out.txt, will have gene, chr, pos, ref, eff, and rsid as new_id.
In the terminal, run:
```{bash, eval=FALSE}
python3 $CODE/annotate_snp_by_position.py \
--input $OUTPUT/pre_weights.txt --chr_col 1 --pos_col 2 \
--lookup_table $DATA/dbSNP150_list.txt.gz --lookup_chr_col 1 --lookup_start_col 2 --lookup_end_col 2 --lookup_newid_col 5 --if_input_has_header 1 \
--out_txtgz $OUTPUT/weights_out.txt.gz
```

# Add varIDs

Read weights_out.txt in R, and define varID from chromosome, position, reference and effect alleles. (The RNA-seq was in hg19, so varID is defined in b37)

```{r}
weights <- fread(glue::glue("{OUTPUT}/weights_out.txt.gz"))
weights$varID <- paste(paste("chr", weights$chromosome, sep = ""), weights$position, weights$ref_allele, weights$eff_allele, "b37", sep = "_")
weights <- weights %>% select(gene, new_id, varID, ref_allele, eff_allele, weight) %>% rename(rsid = new_id)
```

# Make Extra Table

Generate number of snps for each gene from the weights table. For now, include blank columns to match PrediXcan format (gene, genename, n.snps.in.model, pred.perf.R2, pred.perf.pval, pred.perf.qval)

```{r}
extra <- weights %>% group_by(gene) %>% summarise(n.snps.in.model = n())
extra$genename <- NA
extra$pred.perf.R2 <- NA
extra$pred.perf.pval <- NA
extra$pred.perf.qval <- NA
extra <- extra[c(1, 3, 2, 4, 5, 6)]
```

Check distribution of n.snps.in.model for each gene.
```{r}
summary(extra$n.snps.in.model)
hist(extra$n.snps.in.model)
```

# Write to SQLite Database
Create database connection, and write the weights and extra tables to database.
```{r}
model_db = glue::glue("{MODEL}/psychencode_model/psychencode_fusion.db")
conn <- dbConnect(RSQLite::SQLite(), model_db)
dbWriteTable(conn, "weights", weights)
dbWriteTable(conn, "extra", extra)
```
To double check, confirm there is a weights and extra table, and show their contents.
```{r}
dbListTables(conn)
dbGetQuery(conn, 'SELECT * FROM weights') %>% head
dbGetQuery(conn, 'SELECT * FROM extra') %>% head
```
Lastly, disconnect from database connection
```{r}
dbDisconnect(conn)
```

